safe_eth/eth/clients/etherscan_client_v2.py

```diff
@@ -0,0 +1,56 @@
+import os
+from typing import Any, Dict, List, Optional
+from urllib.parse import urljoin
+
+import requests
+
+from safe_eth.eth import EthereumNetwork
+from safe_eth.eth.clients import EtherscanClient
+
+
+class EtherscanClientV2(EtherscanClient):
+    BASE_API_V2_URL = "https://api.etherscan.io"
+
+    def __init__(
+        self,
+        network: EthereumNetwork,
+        api_key: Optional[str] = None,
+        request_timeout: int = int(
+            os.environ.get("ETHERSCAN_CLIENT_REQUEST_TIMEOUT", 10)
+        ),
+    ):
+        super().__init__(EthereumNetwork.MAINNET, api_key, request_timeout)
+        self.network = network
+        self.base_api_url = self.BASE_API_V2_URL
+
+    @classmethod
+    def _get_supported_networks(cls) -> List[Dict[str, Any]]:
```

Comment:
I think this should be cached, and also it shouldn't be a private method

Reply:
IMO this method will be called a few times and may even add a string along the way. It should have no problems without being cached.

It is updated as public method [here](https://github.com/safe-global/safe-eth-py/pull/1407/commits/9b8049300f80f04aa75dd4b1f643d0735ab3d0ed)